front/app/modules/commercial/content_builder/admin/containers/index.tsx

```diff
@@ -21,34 +24,104 @@ import ContentBuilderSettings from '../components/ContentBuilderSettings';
 import { SerializedNodes } from '@craftjs/core';
+import { Locale } from 'typings';
+import eventEmitter from 'utils/eventEmitter';
+
+// intl
+import messages from '../messages';
+import FormattedMessage from 'utils/cl-intl/FormattedMessage';
 
 const StyledRightColumn = styled(RightColumn)`
-  min-height: calc(100vh - ${2 * stylingConsts.menuHeight}px);
+  min-height: calc(100vh - ${stylingConsts.menuHeight}px);
   z-index: 2;
   margin: 0;
   max-width: 100%;
   align-items: center;
   padding-bottom: 100px;
 `;
 
-const ContentBuilderPage = ({ params: { projectId } }: WithRouterProps) => {
+type ContentBuilderErrors = Record<
+  string,
+  { hasError: boolean; selectedLocale: Locale }
+>;
+
+export const ContentBuilderPage = () => {
   const [mobilePreviewEnabled, setMobilePreviewEnabled] = useState(false);
+  const [selectedLocale, setSelectedLocale] = useState<Locale | undefined>();
+  const [draftData, setDraftData] = useState<Record<string, SerializedNodes>>();
+  const { pathname } = useLocation();
+  const { projectId } = useParams() as { projectId: string };
+
   const iframeRef = useRef<HTMLIFrameElement | null>(null);
 
+  const featureEnabled = useFeatureFlag({ name: 'content_builder' });
   const locale = useLocale();
-
+  const locales = useAppConfigurationLocales();
   const contentBuilderLayout = useContentBuilderLayout({
     projectId,
     code: PROJECT_DESCRIPTION_CODE,
   });
 
+  useEffect(() => {
+    if (!isNilOrError(locale)) {
+      setSelectedLocale(locale);
+    }
+  }, [locale]);
+
+  const [contentBuilderErrors, setContentBuilderErrors] =
+    useState<ContentBuilderErrors>({});
+
+  useEffect(() => {
+    const subscription = eventEmitter
+      .observeEvent('contentBuilderError')
+      .subscribe(({ eventValue }) => {
+        setContentBuilderErrors((contentBuilderErrors) => ({
+          ...contentBuilderErrors,
+          ...(eventValue as ContentBuilderErrors),
+        }));
+      });
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+
+  useEffect(() => {
+    const subscription = eventEmitter
+      .observeEvent('deleteContentBuilderElement')
+      .subscribe(({ eventValue }) => {
+        setContentBuilderErrors((contentBuilderErrors) => {
+          const deletedElementId = eventValue as string;
+          const { [deletedElementId]: _deletedElementId, ...rest } =
+            contentBuilderErrors;
+          return rest;
+        });
+      });
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+
+  if (isNilOrError(locales)) {
+    return null;
+  }
+
+  const localesWithError = Object.values(contentBuilderErrors)
+    .filter((node) => node.hasError)
+    .map((node) => node.selectedLocale);
+
   const editorData =
-    !isNilOrError(contentBuilderLayout) && !isNilOrError(locale)
-      ? contentBuilderLayout.data.attributes.craftjs_jsonmultiloc[locale]
+    !isNilOrError(contentBuilderLayout) && selectedLocale
```

Comment:
At this level of nesting, it might be better to use something other than a ternary. It makes the code harder to read.

Reply:
Good point 👍